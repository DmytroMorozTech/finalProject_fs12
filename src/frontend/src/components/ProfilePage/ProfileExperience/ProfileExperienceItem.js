import React from 'react'
import style from './styles'
import CreateIcon from '@material-ui/icons/Create'
import Typography from '@material-ui/core/Typography'
import toggleModalAction from '../../../redux/Modal/modalActions'
import {useDispatch} from 'react-redux'
import {EDIT_EDUCATION} from '../../../redux/Modal/modalTypes'

const ProfileExperienceItem = (props) => {
  const {organization, position, dateStart, dateFinish} = props.workplace
  const classes = style()
  const dash = <span className={classes.dash}>-</span>
  const dispatch = useDispatch()

  return (
    <div className={classes.content}>
      <div className={classes.school}>
        <Typography variant="body1" className={classes.schoolName}>{position}</Typography>
        <Typography>{organization}</Typography>
        <Typography>{dateStart}{dash}{dateFinish}</Typography>
      </div>
      <div className={classes.btnCreate} onClick={() =>
        dispatch(toggleModalAction({modalType: EDIT_EDUCATION}))}>
        <CreateIcon className={classes.createIcon}/>
      </div>
    </div>
  )
}
export default ProfileExperienceItem