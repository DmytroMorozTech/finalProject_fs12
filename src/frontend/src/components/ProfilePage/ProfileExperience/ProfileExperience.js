import React from 'react'
import style from './styles'
import Typography from '@material-ui/core/Typography'
import ProfileExperienceItem from './ProfileExperienceItem'
import {useDispatch} from 'react-redux'
import toggleModalAction from '../../../redux/Modal/modalActions'
import {ADD_EXPERIENCE} from '../../../redux/Modal/modalTypes'
import AddIcon from '@material-ui/icons/Add'

const ProfileExperience = (props) => {
  const workPlaces = props.workPlaces
  const classes = style()
  const dispatch = useDispatch()

  return (
    <div className={classes.root}>
      <div className={classes.header}>
        <Typography variant="h3">
                    Experience
        </Typography>
        <div onClick={() =>
          dispatch(toggleModalAction({modalType: ADD_EXPERIENCE}))}>
          <AddIcon className={classes.createIcon}/>
        </div>
      </div>
      <div>
        {workPlaces && workPlaces.map(workPlace => <ProfileExperienceItem key={workPlace.id} workPlace={workPlace}/>)}
      </div>
    </div>
  )
}

export default ProfileExperience