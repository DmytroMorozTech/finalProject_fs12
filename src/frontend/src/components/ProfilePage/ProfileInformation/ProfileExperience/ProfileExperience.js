import React from 'react'
import style from '../styles'
import Typography from '@material-ui/core/Typography'
import ProfileExperienceItem from './ProfileExperienceItem'
import {useDispatch} from 'react-redux'
import toggleModalAction from '../../../../redux/Modal/modalActions'
import {ADD_NEW_EXPERIENCE} from '../../../../redux/Modal/modalTypes'
import AddIcon from '@material-ui/icons/Add'
import clsx from 'clsx'

const ProfileExperience = (props) => {
  const {workPlaces, isEditable} = props
  const classes = style()
  const dispatch = useDispatch()

  return (
    <div className={classes.root}>
      <div className={classes.header}>
        <Typography variant="h3">
          Experience
        </Typography>
        <div
          className={clsx(!isEditable && classes.hidden)}
          onClick={() =>
            dispatch(toggleModalAction({modalType: ADD_NEW_EXPERIENCE}))}>
          <AddIcon className={classes.createIcon}/>
        </div>
      </div>
      <div>
        {workPlaces &&
        workPlaces
          .sort((wp1, wp2) => wp2.dateStart.localeCompare(wp1.dateStart)) // sorting in DESCENDING ORDER
          .map(workPlace => <ProfileExperienceItem
            key={workPlace.id}
            workPlace={workPlace}
            isEditable = {isEditable}
          />)}
      </div>
    </div>
  )
}

export default ProfileExperience
