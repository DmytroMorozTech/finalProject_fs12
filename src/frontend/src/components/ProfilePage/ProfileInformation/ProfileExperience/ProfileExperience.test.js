import { shallow } from 'enzyme'
import * as redux from 'react-redux'
import React from 'react'
import ProfileExperience from './ProfileExperience'

jest.mock('../styles', () => () => ({}))

const workPlaces = [
  {
    'id': 1,
    'position': 'Chief Operational Officer',
    'responsibilities': 'Quisque id rhoncus mauris. Phasellus quis diam aliquet, gravida massa in, suscipit felis. Aliquam ac turpis arcu. Nunc fermentum dui augue, et aliquam eros ullamcorper imperdiet.',
    'dateStart': '2000-12-31',
    'dateFinish': '2010-06-12',
    'isCurrentlyEmployed': false,
    'organization': {
      'id': 3,
      'name': 'PayPal Holdings Inc.',
      'location': '2211 N 1st St, San Jose, CA 95131, United States',
      'webSite': 'www.paypal.com'
    }
  },
  {
    'id': 2,
    'position': 'CEO',
    'responsibilities': 'Fusce lacinia, nisl et vestibulum ultricies, orci risus luctus urna, et consectetur sem nibh non mauris. Phasellus ac aliquam libero, eu luctus lorem. Nulla tempor risus eu tortor mattis, id eleifend tellus semper. ',
    'dateStart': '2010-06-20',
    'dateFinish': '2013-11-22',
    'isCurrentlyEmployed': false,
    'organization': {
      'id': 2,
      'name': 'Apple',
      'location': '1 Apple Park Way Cupertino, California, 95014-0642 United States',
      'webSite': 'apple.com'
    }
  },
  {
    'id': 3,
    'position': 'Sales manager',
    'responsibilities': 'Phasellus vitae interdum mauris, eget luctus neque. Aenean at venenatis mauris, venenatis convallis velit. Suspendisse hendrerit lectus dolor, sit amet tristique odio lobortis sit amet. Donec ante purus, ',
    'dateStart': '2014-01-20',
    'dateFinish': null,
    'isCurrentlyEmployed': true,
    'organization': {
      'id': 1,
      'name': 'Microsoft',
      'location': 'Redmond, WA, US (HQ) Microsoft Corporation One Microsoft Way',
      'webSite': 'microsoft.com'
    }
  }
]

let isEditable

const setUp = (props) => shallow(<ProfileExperience {...props}/>)

describe('should render ProfileExperience component without props', () => {
  let spyOnUseDispatch
  let mockDispatch

  let emptyWorkPlace

  let component

  beforeEach(() => {
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    emptyWorkPlace = []

    isEditable = false

    component = setUp({emptyWorkPlace, isEditable})
  })
  it('should work dispatch', () => {
    expect(mockDispatch).toHaveBeenCalledTimes(0)
    component.find('div[id=\'toggleModal\']').simulate('click')
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
})

describe('should render ProfileExperience component with props', () => {
  let spyOnUseDispatch
  let mockDispatch

  let component

  beforeEach(() => {
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    isEditable = true

    component = setUp({workPlaces, isEditable})
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
  it('should show ProfileExperienceItem', () => {
    const wrapper = component.find('ProfileExperienceItem')
    expect(wrapper.length).toBe(3)
  })
  it('should work dispatch', () => {
    expect(mockDispatch).toHaveBeenCalledTimes(0)
    component.find('div[id=\'toggleModal\']').simulate('click')
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
})
