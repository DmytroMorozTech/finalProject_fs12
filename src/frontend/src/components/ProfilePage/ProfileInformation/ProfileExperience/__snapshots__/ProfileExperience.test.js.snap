// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ProfileExperience component with props to match snapshot 1`] = `
<div>
  <div>
    <WithStyles(ForwardRef(Typography))
      variant="h3"
    >
      Experience
    </WithStyles(ForwardRef(Typography))>
    <div
      className=""
      id="toggleModal"
      onClick={[Function]}
    >
      <Memo(ForwardRef(AddIcon)) />
    </div>
  </div>
  <div>
    <ProfileExperienceItem
      isEditable={true}
      key="3"
      workPlace={
        Object {
          "dateFinish": null,
          "dateStart": "2014-01-20",
          "id": 3,
          "isCurrentlyEmployed": true,
          "organization": Object {
            "id": 1,
            "location": "Redmond, WA, US (HQ) Microsoft Corporation One Microsoft Way",
            "name": "Microsoft",
            "webSite": "microsoft.com",
          },
          "position": "Sales manager",
          "responsibilities": "Phasellus vitae interdum mauris, eget luctus neque. Aenean at venenatis mauris, venenatis convallis velit. Suspendisse hendrerit lectus dolor, sit amet tristique odio lobortis sit amet. Donec ante purus, ",
        }
      }
    />
    <ProfileExperienceItem
      isEditable={true}
      key="2"
      workPlace={
        Object {
          "dateFinish": "2013-11-22",
          "dateStart": "2010-06-20",
          "id": 2,
          "isCurrentlyEmployed": false,
          "organization": Object {
            "id": 2,
            "location": "1 Apple Park Way Cupertino, California, 95014-0642 United States",
            "name": "Apple",
            "webSite": "apple.com",
          },
          "position": "CEO",
          "responsibilities": "Fusce lacinia, nisl et vestibulum ultricies, orci risus luctus urna, et consectetur sem nibh non mauris. Phasellus ac aliquam libero, eu luctus lorem. Nulla tempor risus eu tortor mattis, id eleifend tellus semper. ",
        }
      }
    />
    <ProfileExperienceItem
      isEditable={true}
      key="1"
      workPlace={
        Object {
          "dateFinish": "2010-06-12",
          "dateStart": "2000-12-31",
          "id": 1,
          "isCurrentlyEmployed": false,
          "organization": Object {
            "id": 3,
            "location": "2211 N 1st St, San Jose, CA 95131, United States",
            "name": "PayPal Holdings Inc.",
            "webSite": "www.paypal.com",
          },
          "position": "Chief Operational Officer",
          "responsibilities": "Quisque id rhoncus mauris. Phasellus quis diam aliquet, gravida massa in, suscipit felis. Aliquam ac turpis arcu. Nunc fermentum dui augue, et aliquam eros ullamcorper imperdiet.",
        }
      }
    />
  </div>
</div>
`;
